web/src/main/java/org/cbioportal/web/DataAccessTokenController.java

```diff
@@ -1,98 +1,97 @@
- * License.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Affero General Public License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+* This file is part of cBioPortal.
+*
+* cBioPortal is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Affero General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 package org.cbioportal.web;
 
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
 import org.cbioportal.model.DataAccessToken;
 import org.cbioportal.service.DataAccessTokenService;
-import org.cbioportal.service.DataAccessTokenServiceFactory;
 import org.cbioportal.service.exception.DataAccessTokenNoUserIdentityException;
 import org.cbioportal.service.exception.DataAccessTokenProhibitedUserException;
-import org.cbioportal.service.impl.UnauthDataAccessTokenServiceImpl;
 import org.cbioportal.web.config.annotation.InternalApi;
-
-import io.swagger.annotations.*;
-import java.util.*;
-import javax.annotation.PostConstruct;
-import org.springframework.beans.factory.annotation.*;
-import org.springframework.http.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.security.core.Authentication;
 import org.springframework.validation.annotation.Validated;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.client.HttpClientErrorException;
 
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiParam;
+
 @InternalApi
-@RestController
+@RequestMapping // replaces @RestController; controller is created conditionally in DataAccessTokenControllerConfig
```

Comment:
It looks like the removal of @RestController causes the following DataAccessTokenControllerTest.revokeAllTokensForUserTest() failure:

`
Running org.cbioportal.web.DataAccessTokenControllerTest
Tests run: 7, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 17.534 sec <<< FAILURE!
revokeAllTokensForUserTest(org.cbioportal.web.DataAccessTokenControllerTest)  Time elapsed: 0.772 sec  <<< ERROR!
javax.servlet.ServletException: Circular view path [data-access-tokens]: would dispatch back to the current handler URL [/data-access-tokens] again. Check your ViewResolver setup
! (Hint: This may be the result of an unspecified view, due to default view name generation.)      
`

After some digging, I think its because @RestController included @ResponseBody which is required to properly unmarshal into JSON.  Adding @ResponseBody after @RequestMapping should solve this issue.

Reply:
PR was updated with this solution.